projects/output-and-error-handling/main.go

```diff
@@ -0,0 +1,61 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+)
+
+func main() {
+	for {
+		resp, err := http.Get("http://localhost:8080")
+		if err != nil {
+			fmt.Println("Server not working at the moment!")
+			fmt.Fprint(os.Stderr, "Server not working at the moment! ", err, "\n")
+			return
+		}
+		defer resp.Body.Close()
+
+		if resp.StatusCode == 429 {
+			// fmt.Printf("\n%#v\n", resp.Header)
+			fmt.Println("Please wait!")
+			retryAfterStr := resp.Header["Retry-After"][0]
+			// retryAfterNum, numErr := strconv.Atoi(retryAfterStr)
+			retryAfterNum, numErr := strconv.ParseFloat(retryAfterStr, 64)
+
+			var retryDuration time.Duration
+			if numErr != nil {
+				result, timeErr := time.Parse(http.TimeFormat, retryAfterStr)
+				if timeErr != nil {
+					fmt.Fprint(os.Stderr, "Retry after a while (1s): ", timeErr, "\n")
+					result = time.Now().Add(time.Second)
+				}else{
+					fmt.Fprint(os.Stderr, "Retry after a time stamp: ", retryAfterStr, "\n")
+				}
+				retryDuration = -time.Since(result)
+			} else {
+				fmt.Fprint(os.Stderr, "Retry after ", retryAfterNum, "s \n")
+				retryDuration = time.Duration(float64(time.Second) * retryAfterNum)
+				fmt.Println(retryAfterNum,retryDuration)
+
+			}
+			// fmt.Println(retryDuration)
+			time.Sleep(retryDuration)
+
+		} else if resp.StatusCode == 200 {
+
+			scanner := bufio.NewScanner(resp.Body)
+			for i := 0; scanner.Scan() && i < 5; i++ {
```

Comment:
What does the 5 here mean? How did you pick it? Why was that a good value to use?

Reply:
I picked it up from here:     https://gobyexample.com/http-client
Now I read the explanation and means `Print the first 5 lines of the response body`